### PR TITLE
Add info on Brain Board Debugging

### PR DESCRIPTION
**Pull request by trickeydan**

### CODE REVIEW

File: docs/kit/software/software-debugging.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- `udevadm monitor` - Monitor udev events
+- `journalctl -f` - View and follow journal logs
+- `journalctl -fu astdiskd` - View and follow astdiskd logs
+- `journalctl -fu astprocd` - View and follow astprocd logs
+- `journalctl -fu astmetad` - View and follow astmetad logs
+- `journalctl -fu astwifid` - View and follow astwifid and hostapd logs
+- `journalctl -fu kchd` - View and follow astwifid and hostapd logs
+- `journalctl -fu servohack` - View and follow astwifid and hostapd logs
+- `cat /var/lib/misc/dnsmasq.leases` - List DHCP leases on robot WiFi
+- `mosquitto_sub -v -t "astoria/#"` - Monitor MQTT Event Bus messages
```
Comment (WillB97) [inline docs/kit/software/software-debugging.md:32]: The image doesn't look to have `mosquitto_sub` included.

File: docs/kit/software/software-debugging.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- `astctl usercode restart` - Restart any running usercode
+- `astctl list-disks` - List inserted USB drives and their types
+- `astctl metadata show` - Show all current metadata
+- `astctl metadata set <key> <val>` - Set a metadata attribute
+- `cat /var/srobo/cache/astmetad-metadata.json` - View the metadata cache
+
+### Misc
+
+- `nft -f /etc/nftables` - Reload firewall
+- `nano /etc/nftables` - Edit the firewall configuration
```
Comment (WillB97) [inline docs/kit/software/software-debugging.md:49]: /etc/firewalls is a directory here which causes nano to throw an error.

Comment (trickeydan) [reply docs/kit/software/software-debugging.md]: 8c51fed

File: docs/kit/software/software-debugging.md (hunk 1 above)
Comment (trickeydan) [reply docs/kit/software/software-debugging.md]: 8c51fed

File: docs/kit/software/software-debugging.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+- `dmesg -w` - View and follow the kernel log
+- `udevadm monitor` - Monitor udev events
+- `journalctl -f` - View and follow journal logs
+- `journalctl -fu astdiskd` - View and follow astdiskd logs
+- `journalctl -fu astprocd` - View and follow astprocd logs
+- `journalctl -fu astmetad` - View and follow astmetad logs
+- `journalctl -fu astwifid` - View and follow astwifid and hostapd logs
+- `journalctl -fu kchd` - View and follow astwifid and hostapd logs
+- `journalctl -fu servohack` - View and follow astwifid and hostapd logs
```
Comment (raccube) [inline docs/kit/software/software-debugging.md:30]: These don't match up.

File: docs/kit/software/software-debugging.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+- `cat /var/lib/misc/dnsmasq.leases` - List DHCP leases on robot WiFi
+
+### Astoria Specific Commands
+
+- `astctl usercode show` - Show information about usercode
```
Comment (raccube) [inline docs/kit/software/software-debugging.md:35]:
```suggestion
- `astctl usercode show` - Show the state of usercode (running, crashed, etc. as well as the location of usercode)
```

Comment (raccube) [inline docs/kit/software/software-debugging.md:35]:
```suggestion
- `astctl usercode show` - Show information about usercode (such as location of usercode, state e.g. running or crashed)
```

**Review by raccube — CHANGES_REQUESTED** [review]

File: docs/kit/software/ethernet.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- Select type `ethernet`.
+- Choose an appropriate name, e.g `robot-ethernet` and select create.
+- Navigate to the IPv4 Settings tab.
+- Select Method `Shared to other computers`.
+- Start the connection from the UI, or by running `nmcli connection up robot-ethernet`
+
+#### MacOS
+
+- Open System Preferences.
+- Open the Sharing preference pane and select Internet Sharing in the left-hand Service list.
```
Comment (raccube) [inline docs/kit/software/ethernet.md:50]: In System Settings (13.0+) this is under General -> Sharing iirc.

File: docs/kit/software/software-debugging.md (hunk 3 above)
Comment (trickeydan) [reply docs/kit/software/software-debugging.md:30]: Match up?

Comment (raccube) [reply docs/kit/software/software-debugging.md:30]: Don’t think servohack or kchd have output logs for astwifid/hostapd

File: docs/kit/software/ethernet.md (hunk 5 above)
Comment (raccube) [reply docs/kit/software/ethernet.md:50]: image

**Pull request merged**